src/Node/TemplateBoxNode.php

```diff
@@ -50,14 +83,87 @@ public function compile(Compiler $compiler): void
 
         $value = $this->getNode('message')->getAttribute('value');
 
-        $message = <<<CODE
+        if (null !== $this->translator) {
+            $translationBundle = null;
+
+            if ($this->hasNode('translationBundle')) {
+                $translationBundle = $this->getNode('translationBundle');
+            }
+
+            if ($translationBundle) {
+                $translationBundle = $translationBundle->getAttribute('value');
+            }
+
+            $message = <<<CODE
+"<div class='alert alert-default alert-info'>
+    <strong>{$this->translator->trans($value, [], $translationBundle)}</strong>
+    <div>{$this->translator->trans('sonata_core_template_box_file_found_in', [], 'SonataCoreBundle')} <code>{\$this->getTemplateName()}</code>.</div>
+</div>"
+CODE;
+        } else {
+            $message = <<<CODE
 "<div class='alert alert-default alert-info'>
     <strong>{$value}</strong>
     <div>This file can be found in <code>{\$this->getTemplateName()}</code>.</div>
 </div>"
 CODE;
+        }
 
         $compiler
             ->write("echo $message;");
     }
+
+    /**
+     * @deprecated since sonata-project/twig-extensions 0.x, to be removed with 1.0
+     *
+     * @param AbstractExpression      $message           Node message to display
+     * @param AbstractExpression|null $translationBundle Node translation bundle to use for display
+     * @param int                     $enabled           Is Symfony debug enabled?
+     * @param string|null             $lineno            Symfony template line number
+     * @param string|null             $tag               Symfony tag name
+     */
+    private function deprecatedConstructor(
+        AbstractExpression $message,
+        ?AbstractExpression $translationBundle = null,
+        $enabled,
+        TranslatorInterface $translator,
+        $lineno,
+        ?string $tag = null
+    ) {
+        @trigger_error(
+            sprintf(
+                'The translator dependency in %s is deprecated since 0.x and will be removed in 1.0. '.
```

Comment:
IMO, we should be more specific about "translator dependency", since it's not clear if we are referencing a package dependency or an argument required by this class.